File: shell-examples/src/main/js/io/vertx/example/shell/command/custom_command.js
var Command = require("vertx-shell-js/command");
var ShellService = require("vertx-shell-js/shell_service");

var command = Command.builder("my-command").processHandler(function (process) {
  process.write("hello sir\n").end();
}).build();

var service = ShellService.create(vertx, {
  "telnetOptions" : {
    "host" : "localhost",
    "port" : 3000
  }
});
service.getCommandRegistry().registerCommand(command);
service.start();

File: shell-examples/src/main/rb/io/vertx/example/shell/command/custom_command.rb
require 'vertx-shell/command'
require 'vertx-shell/shell_service'

command = VertxShell::Command.builder("my-command").process_handler() { |process|
  process.write("hello sir\n").end()
}.build()

service = VertxShell::ShellService.create($vertx, {
  'telnetOptions' => {
    'host' => "localhost",
    'port' => 3000
  }
})
service.get_command_registry().register_command(command)
service.start()
